### PR TITLE
Added reuseable workflows

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Added reuseable workflow in place of individual WF 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by evalsocket** [comment]
@EngHabu do you know why it is failing https://github.com/flyteorg/datacatalog/runs/5175386064?check_suite_focus=true ?

**Review by EngHabu — APPROVED** [review]
